src/main/scala/com/yotpo/metorikku/output/writers/jdbc/JDBCQueryWriter.scala

```diff
@@ -0,0 +1,46 @@
+package com.yotpo.metorikku.output.writers.jdbc
+
+import java.sql.{Date, DriverManager}
+
+import com.yotpo.metorikku.configuration.outputs.JDBC
+import com.yotpo.metorikku.output.MetricOutputWriter
+import org.apache.spark.sql.DataFrame
+
+import scala.collection.mutable
+
+class JDBCQueryWriter(props: Map[String, String], config: Option[JDBC]) extends MetricOutputWriter {
+
+  case class JDBCQueryProperties(query: String)
+
+  val options = JDBCQueryProperties(props("query"))
+
+  override def write(dataFrame: DataFrame): Unit = {
+    config match {
+      case Some(config) =>
+        dataFrame.foreachPartition(partition => {
+          val conn = DriverManager.getConnection(config.connectionUrl, config.user, config.password)
+          val stmt = conn.prepareStatement(options.query)
+
+          partition.foreach(row => {
+            for (i <- 1 to row.size) {
```

Comment:
So you're depending on the order of the columns, I'm wondering if perhaps there is a better way, by the column names perhaps?

Reply:
The postgres (and SQL as well) prepare statements accept parameters by order. We can get from the array of column names and push the value by the order he provided. That seems like an overhead while the user should know about the order of the column because he wrote the previous "Select" step.